src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -3708,6 +3710,58 @@ private static bool TryMapParameter((SyntaxNode? Node, int Ordinal) parameterKey
             }
         }
 
+        protected virtual void ReportLambdaAttributeRudeEdits(SemanticModel oldModel,
+            SyntaxNode oldLambdaBody,
+            SemanticModel newModel,
+            SyntaxNode newLambdaBody,
+            List<RudeEditDiagnostic> diagnostics,
+            bool hadSignatureEdits,
+            CancellationToken cancellationToken)
+        {
+            var newLambda = GetLambda(newLambdaBody);
+            var oldLambda = GetLambda(oldLambdaBody);
+
+            Debug.Assert(IsNestedFunction(newLambda) == IsNestedFunction(oldLambda));
+
+            // queries are analyzed separately
+            if (!IsNestedFunction(newLambda))
+            {
+                return;
+            }
+
+            var oldLambdaSymbol = GetLambdaExpressionSymbol(oldModel, oldLambda, cancellationToken);
+            var newLambdaSymbol = GetLambdaExpressionSymbol(newModel, newLambda, cancellationToken);
+
+            var reportDiagnostic = false;
+            if (!oldLambdaSymbol.GetAttributes().SequenceEqual(newLambdaSymbol.GetAttributes()) ||
+                !oldLambdaSymbol.GetReturnTypeAttributes().SequenceEqual(newLambdaSymbol.GetReturnTypeAttributes()))
+            {
+                reportDiagnostic = true;
+            }
+
+            // If the old and new signatures are the same then we are safe to compare parameter attributes more easily
+            if (!hadSignatureEdits)
+            {
+                for (var i = 0; i < oldLambdaSymbol.Parameters.Length; i++)
+                {
+                    if (!oldLambdaSymbol.Parameters[i].GetAttributes().SequenceEqual(newLambdaSymbol.Parameters[i].GetAttributes()))
+                    {
+                        reportDiagnostic = true;
+                        break;
+                    }
+                }
+            }
+
+            if (reportDiagnostic)
+            {
+                diagnostics.Add(new RudeEditDiagnostic(
```

Comment:
This will report "Updating local function will prevent the debugging session from continuing" and squiggle the local function. That's overly generic/imprecise, since some updates to local function are allowed.
It'd be better to squiggle the specific attribute that doesn't match.

Reply:
Before I go and write one, is there a helper also that compares sequences and also returns the item that differs?

Or perhaps just squiggling the local function name is sufficient?

Reply:
I don't remember if we have such a helper. Look around a bit and if you can't find it feel free to add one.
In general we try to be as specific as we can (within reason). Makes the experience much better.